packages/pwa/app/hooks/use-site.js

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {convertToFullyQualifiedUrl} from '../utils/utils'
+import {useLocation} from 'react-router-dom'
+import {resolveSiteFromUrl} from '../utils/site-utils'
+
+const useSite = () => {
+    const {pathname, search} = useLocation()
+    const url = convertToFullyQualifiedUrl(`${pathname}${search}`)
+    const site = resolveSiteFromUrl(url)
```

Comment:
In the grand scheme of things the `url` doesn't really change often (especially with respect to component renders/re-renders).So when you are doing a unit of work (in this case determining the site from the url) it only makes sense to do that when the url changes. You've probably used `useMemo` before, but it ight be a good idea to do that here. Something like:

```
let site
useMemo(() => {
   site = resolveSiteFromUrl(pathToUrl(`${pathname}${search}`))
}, location)
```

This code will ensure that you only run the site resolution algorithm when the url changes.

On a site note, when I gave you the example for the name `convertToFullyQualifiedUrl` I said it was probably a really bad name, and it in fact is. There isn't really suck thing as a fully qualified url. So thats why I changed the above name to `pathToUrl`.